db/new_seeds/base/add_declarations.rb

```diff
@@ -1,72 +1,209 @@
+
+        # Create all needed milestones in a single transaction
+        if needed_milestone_types.any?
+          ActiveRecord::Base.transaction do
+            needed_milestone_types.each do |declaration_type|
+              next if schedule.milestones.exists?(declaration_type:)
+
+              FactoryBot.create(
+                :milestone,
+                schedule:,
+                declaration_type:,
+                start_date: 1.day.ago,
+              )
+            end
+          end
+        end
+
+        # Load all milestones for this schedule to avoid repeated queries
+        schedule_milestones = schedule.milestones.index_by(&:declaration_type)
+
+        # Process declarations for each profile
+        profiles.each do |participant_profile|
+          # Determine valid states for this profile
+          profile_states = states.dup
+          fundable_states = %w[payable paid awaiting_clawback clawed_back]
+
+          unless participant_profile.fundable?
+            profile_states.reject! { |state| state.in?(fundable_states) }
+          end
+
+          # Filter declaration types for this profile
+          profile_declaration_types = declaration_types.dup
+          existing_types = participant_profile.participant_declarations.pluck(:declaration_type)
+          profile_declaration_types.reject! { |type| type.in?(existing_types) }
+
+          if participant_profile.mentor? && schedule.cohort.mentor_funding?
+            profile_declaration_types.reject! { |type| type.match?(/retained|extended/) }
+          end
+
+          unless schedule.schedule_identifier.match?(/extended/)
+            profile_declaration_types.reject! { |type| type.match?(/extended/) }
+          end
+
+          # Prepare declarations to create
+          declarations_to_create = []
+
+          profile_declaration_types.each do |declaration_type|
+            # Keep creating declarations with a 50% chance
+            break unless Faker::Boolean.boolean(true_ratio: 0.5)
+
+            # Get milestone for this declaration type
+            milestone = schedule_milestones[declaration_type]
+            next unless milestone # Skip if no milestone found
+
+            # Determine state for this declaration
+            state = profile_states.sample
+
+            # Add to declarations to create
+            declarations_to_create << {
+              declaration_type:,
+              participant_profile:,
+              cpd_lead_provider:,
+              cohort: schedule.cohort,
+              milestone_date: milestone.start_date,
+              state:,
+              profile_type: participant_profile.ect? ? :ect : :mentor,
+            }
+          end
+
+          # Create declarations in a single transaction
+          next unless declarations_to_create.any?
+
+          ActiveRecord::Base.transaction do
+            declarations_to_create.each do |declaration_data|
+              milestone_date = declaration_data.delete(:milestone_date)
```

Comment:
is there a reason we're deleting this data?

Reply:
Hi @cwrw It's just deleting from the hash which goes as a param to the factory and using the value to `travel_to`